### ISSUE TITLE
Trouble getting the mnist example to converge

### ISSUE THREAD

**Issue by kaihuchen** [carpedm20/DCGAN-tensorflow#55]
I have not been able to get the mnist example to converge. Even after days of training (8-CPU system, no GPU) following the standard mnist example in the kit with no change (run as: python main.py --dataset mnist --is_train True), the d_loss/g_loss values just bounced around 1.3/0.7 with little sign of getting any better. I have not observed any error messages so far.

Is it expected that the mnist example will require a lot more computing power than I have in order to make it converge? If there are some hidden problems then what should I look for?

**Comment by EncodeTS** [comment]
I meet the same problem,it just can't converge.
But the picture in the **samples** folder are getting more and more clear.

**Comment by ppwwyyxx** [comment]
for GAN, however, convergence is likely an indication of failure...

**Comment by carpedm20** [comment]
I suggest you not to use my code but just a reference code. This code is one of my earliest practice with TensorFlow and now there are "too" many good papers and its implementation in *generative model* fields which published and open-sourced after DCGAN is appeared.

- https://github.com/openai/improved-gan/
- https://github.com/openai/pixel-cnn
- https://github.com/openai/iaf
- https://github.com/openai/InfoGAN
- and literally a lot and it's hard to find...

**Comment by auroua** [comment]
What's the algorithms used to generate MNIST digitals? I found a lot of source programs using label information, that original paper not contains.
```
            h0 = tf.nn.relu(self.g_bn0(linear(z, self.gfc_dim, 'g_h0_lin')))
            h0 = tf.concat(1, [h0, y])

            h1 = tf.nn.relu(self.g_bn1(linear(h0, self.gf_dim*2*s4*s4, 'g_h1_lin')))
            h1 = tf.reshape(h1, [self.batch_size, s4, s4, self.gf_dim * 2])
```

**Comment by zhangqianhui** [comment]
@auroua  Yes , just like "conditional adversarial networks"
Adding label to training data will makes result realistic in general.

**Comment by zhangqianhui** [comment]
the loss  just bounced around 1.3/0.7 means "converge" in my experiments.

**Comment by kaihuchen** [comment]
Recently I did a [whole bunch of experiments](http://www.terraai.org/)  using [cGAN](https://github.com/phillipi/pix2pix), such a getting it to  convert from black-and-white portraits to color ones, learning monocular depth perception, or performing image operations such as erasing the background. But eventually I ran into problem when trying to use cGAN's z vector as a neural image model, which I intend to apply to videos. The problem I had with cGAN is that its U-Net architecture is great for image-to-image mapping, but it also weakens the z vector as a latent representation. So for this purpose I think I need to go back to use DCGAN instead. Just want to point out here that DCGAN is by no means obsolete.
